app/src/main/java/com/example/habitsmasher/FollowUserDialog.java

```diff
@@ -109,18 +118,23 @@ public void onComplete(@NonNull Task<QuerySnapshot> task) {
                                             displayErrorMessage(CANNOT_FOLLOW_YOURSELF_ERROR);
                                             return;
                                         }
+                                        // if the user requests to follow someone they have already
+                                        // requested to follow
+                                        if (searchInRequestsList(userToFollow.getId(),currentUserId)) {
+                                            displayErrorMessage(ALREADY_REQUESTED_TO_FOLLOW_USER);
+                                            return;
+                                        }
 
-                                        userToFollow.addNewFollower(currentUserId);
-
-                                        // add new follower to the user to follow
-                                        addNewFollowerForUserInDatabase(userToFollow.getId(),
-                                                                        currentUserId);
-
-                                        // add to new followed user to following of current user
-                                        addUserToFollowingForUserInDatabase(currentUserId,
-                                                                            userToFollow.getId());
+                                        // if the user is already following the requested user
+                                        if (searchInFollowerList(userToFollow.getId(), currentUserId)) {
+                                            displayErrorMessage(ALREADY_FOLLOWING);
+                                            return;
+                                        }
```

Comment:
Simply added two checks to the requesting process that prevents a user from requesting to follow a user that have either a) already sent a request to or b) is already following